pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -626,9 +627,15 @@ public void setObject(int parameterIndex, Object in, int targetSqlType, int scal
         setString(parameterIndex, castToString(in), Oid.BPCHAR);
         break;
       case Types.VARCHAR:
-      case Types.LONGVARCHAR:
         setString(parameterIndex, castToString(in), getStringType());
         break;
+      case Types.LONGVARCHAR:
+        if (in instanceof InputStream) {
+          preparedParameters.setText(parameterIndex, (InputStream)in);
```

Comment:
For future reference, I duplicated the problem in this particular travis-ci instance by
1) changing to Postgres 9.4 as the test server
2) setting the environment variables (on Windows - set QUERY_MODE=simple ...)
3) running "mvn clean package -B -V -DprefereQueryMode=simple -P release,coverage"
4) I was already using Oracle JDK 1.8 and did not need to change it.
5) Operating system doesn't seem to matter as the test fails on Windows as well as Linux.
To find the values for 2 and 3, click the details link next to the Travis CI failure. Click on the failed  build job link. The environment variables for the failed job are near line 150 of the debug log. The mvn command is near line 250.